src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -94,5 +100,33 @@ private IVsReportExternalErrors GetExternalErrorReporter(ProjectId projectId, st
 
             return new ProjectExternalErrorReporter(projectId, errorCodePrefix, _serviceProvider);
         }
+
+        private static void EnsurePackageLoaded(IServiceProvider serviceProvider, string language)
+        {
```

Comment:
Currently, CPS invokes CreateProjectContext after switching to UI thread, but I am not sure about other clients (DPL, F#, etc.).
Post Dev15-RTM, plan is to move CPS to also invoke this from background thread, at which point, we should introduce a separate Initialize API on the IWorkspaceProjectFactory, which would be required to be invoked from UI thread.